libraries/plugins/custom_operations/include/graphene/custom_operations/custom_operations_plugin.hpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2019 oxarbitrage and contributors.
+ *
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#pragma once
+
+#include <graphene/app/plugin.hpp>
+#include <graphene/chain/database.hpp>
+
+#include <graphene/custom_operations/custom_objects.hpp>
+#include <graphene/custom_operations/custom_operations.hpp>
+#include <graphene/custom_operations/custom_evaluators.hpp>
+
+namespace graphene { namespace custom_operations {
+using namespace chain;
+
+namespace detail
+{
+    class custom_operations_plugin_impl;
+}
+
+class custom_operations_plugin : public graphene::app::plugin
+{
+   public:
+      custom_operations_plugin();
+      virtual ~custom_operations_plugin();
+
+      std::string plugin_name()const override;
+      std::string plugin_description()const override;
+      virtual void plugin_set_program_options(
+         boost::program_options::options_description& cli,
+         boost::program_options::options_description& cfg) override;
+      virtual void plugin_initialize(const boost::program_options::variables_map& options) override;
+      virtual void plugin_startup() override;
+
+      friend class detail::custom_operations_plugin_impl;
+      std::unique_ptr<detail::custom_operations_plugin_impl> my;
+};
+
+typedef fc::static_variant<account_storage_map, account_storage_list> custom_plugin_operation;
+
+struct custom_operation_wrapper {
+   uint8_t unused_data; // if first char of custom_op.data is 0xFF we unpack, this char is not used anymore then.
```

Comment:
What's the purpose of this then?

Reply:
The data contained on a custom operation must start with this character in order to be processed by the plugin. This is to avoid other kind of data in custom operation created by another plugin or whatever to be processed. We don't want to act in all custom operations but just in the ones that were specifically created for the purpose of being unpacked here.

After we find this character is in the first positions of the data we don't need it anymore. The wrapper was created to do this instead of deleting or other operations that will affect the performance.

Reply:
Since the presence of the 0xff byte doesn't guarantee that the data is valid in the sense we expect, I think it would be simpler to leave it out. Just try to deserialize and ignore if it fails.

Reply:
I *thought* there would be fewer failures with the 0xff byte.

Reply:
Yes, but these failures don't hurt and we need to deal with them anyway.